Base/Models/Public/Include/Models/UnifyedModel.hpp

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <map>
+#include <vector>
+#include <string>
+
+namespace Models
+{
+using FieldNames = std::vector<std::string>;
+
+template <typename TKey>
+class Comparator
+{
+public:
+    bool operator()(const TKey& first, const TKey& second) const noexcept
+    {
+        return static_cast<size_t>(first) < static_cast<size_t>(second);
+    }
+};
+
+template <typename TKey, typename TData = std::string, typename Comp = Comparator<TKey>>
+using Map = std::map<TKey, TData, Comp>;
+
+
+template <class TEnum,class TResult>
+class UnifiedModel
+{
+public:
+    UnifiedModel(const std::string& modelName, size_t amountFields): _modelName(modelName), _amountOfFields(amountFields)
+    {
+    }
+
+    const std::string& getModelName() const noexcept { return _modelName; }
+
+    const std::string& fieldName(TEnum anyEnum) const noexcept
+    {
+        return _fieldData[anyEnum];
+    }
+
+    virtual void fill(const TResult& response) const = 0;  
+
+    std::string& operator[](TEnum anyEnum) const { return _data[anyEnum]; }
+   
+    virtual ~UnifiedModel() = default;
+
+protected:
+    void init(const FieldNames& fieldNames)
+    {
+        size_t counter{0};
+        while (counter < _amountOfFields)
+        {
+            _fieldData.insert(std::pair<TEnum, std::string>(this->getNumEnum(counter),
+                                                                  *(fieldNames.begin() + counter)));
+
+            _data.insert(std::pair<TEnum, std::string>(this->getNumEnum(counter), std::string{}));
+            ++counter;
+        }
+    }
+
+    TEnum toEnum(const std::string& fieldName) const
+    {
+        return std::find_if(_fieldData.begin(), _fieldData.end(), [&fieldName](auto pair)
+                            {
+                                if (pair.second == fieldName)
+                                    return true;
+                                else
+                                    return false;
+                            })->first;
+    }
+
+private:
+    /*
+     * @method getNumEnum
+     * @param size_t num -> used to find all objects from specific enum class
+     * @returns Object of specific enum class
+     */
+    virtual TEnum getNumEnum(size_t num) const = 0;
+  
+private:
+    std::string              _modelName;
+    size_t                   _amountOfFields;
+
+protected:
```

Comment:
Align protected field with the one on line 46

Reply:
I separate methods and fields, if I understand u correctly